onnx/test/reference_evaluator_ml_test.py

```diff
@@ -0,0 +1,1414 @@
+                check_model(onx)
+                x = np.arange(6).reshape((-1, 2)).astype(np.float32)
+                self._check_ort(onx, {"X": x}, rev=True, atol=1e-4)
+                sess = ReferenceEvaluator(onx)
+                got = sess.run(None, {"X": x})
+                expected = expected_post[post]
+                assert_allclose(expected[1], got[1], atol=1e-4)
+                assert_allclose(expected[0], got[0])
+
+    @unittest.skipIf(not ONNX_ML, reason="onnx not compiled with ai.onnx.ml")
+    def test_linear_classifier_binary(self):
+        X = make_tensor_value_info("X", TensorProto.FLOAT, [None, None])
+        In = make_tensor_value_info("I", TensorProto.INT64, [None])
+        Y = make_tensor_value_info("Y", TensorProto.FLOAT, [None, None])
+        expected_post = {
+            "NONE": [
+                np.array([1, 1], dtype=np.int64),
+                np.array([[-9.53, 9.53], [-6.65, 6.65]], dtype=np.float32),
+            ],
+            "LOGISTIC": [
+                np.array([1, 1], dtype=np.int64),
+                np.array(
+                    [[7.263436e-05, 9.999274e-01], [1.292350e-03, 9.987077e-01]],
+                    dtype=np.float32,
+                ),
+            ],
+            "SOFTMAX": [
+                np.array([1, 1], dtype=np.int64),
+                np.array(
+                    [[5.276517e-09, 1.000000e00], [1.674492e-06, 9.999983e-01]],
+                    dtype=np.float32,
+                ),
+            ],
+            "SOFTMAX_ZERO": [
+                np.array([1, 1], dtype=np.int64),
+                np.array(
+                    [[5.276517e-09, 1.000000e00], [1.674492e-06, 9.999983e-01]],
+                    dtype=np.float32,
+                ),
+            ],
+        }
+        x = np.arange(6).reshape((-1, 3)).astype(np.float32)
+        for post in ["SOFTMAX", "NONE", "LOGISTIC", "SOFTMAX_ZERO"]:
+            expected = expected_post[post]
+            with self.subTest(post_transform=post):
+                node1 = make_node(
+                    "LinearClassifier",
+                    ["X"],
+                    ["I", "Y"],
+                    domain="ai.onnx.ml",
+                    classlabels_ints=[0, 1],
+                    coefficients=[-0.58, -0.29, -0.09],
+                    intercepts=[10.0],
+                    multi_class=0,
+                    post_transform=post,
+                )
+                graph = make_graph([node1], "ml", [X], [In, Y])
+                onx = make_model(graph, opset_imports=OPSETS)
+                check_model(onx)
+                # onnxruntime answer seems odd.
+                # self._check_ort(onx, {"X": x}, rev=True)
+                sess = ReferenceEvaluator(onx)
+                got = sess.run(None, {"X": x})
+                assert_allclose(expected[1], got[1], atol=1e-6)
+                assert_allclose(expected[0], got[0])
+
+    @unittest.skipIf(not ONNX_ML, reason="onnx not compiled with ai.onnx.ml")
```

Comment:
I wonder if the whole class can be skipped conditionally to make things simpler

Reply:
I did not write a test checking the runtime does not work in case the flag is not present. But this way of doing makes it still possible.